appcelerator-archive/amp
# Stack network 3

related to #313 
Use external network in stack, using syntaxes:

```
services:
  myservice:
     networks:
        anexternalnetwork
networks:
   anexternalnetwork:
     external: true
```

or 

```
services:
  myservice:
     networks:
        myexternalnetwork
networks:
   myexternalnetwork:
     external:
        name: anexternalnetwwork
```

two tests has been added, sample-12.ym for the first syntax, sample-13.yml for the second.

test: (seems long, but fast to test in fact)
- stop amp: sudo ./swarm stop
- rebuild amp: make install
- start amp: sudo ./swarm start
- launch amplifier locally

first syntax test:
- create stack test1: amp stack up test1 -f api/rpc/stack/test_samples/sample-12.yml
- search test1-pinger container id:  docker ps | grep test1-pinger
- inspect the container: docker inspect [containerId]
- see that the container is attached to amp-infra in the networks section
- search  haproxy: docker ps | grep haproxy
- enter in HAProxy container:  docker exec -it [haproxyId] sh
- in the container verify that we can ping the service: ping ext1-pinger
- It shows that the ext1-pinger service is well attached to the amp-infra network as an external 

second syntax test:
- create stack test2: amp stack up test2 -f api/rpc/stack/test_samples/sample-13.yml
- search test2-pinger container id:  docker ps | grep test2-pinger
- inspect the container: docker inspect [containerId]
- see that the container is attached to amp-infra in the networks section
- to test the second syntax for an external network:
- re-enter in HAProxy container:  docker exec -it [haproxyId] sh
- in the container verify that we can ping the service: ping ext2-pinger
- It shows that the ext2-pinger service is well attached to the amp-infra network as an external 

regression test:
- make test
